### DISCUSSION TITLE
How can I traverse the pore network of a medium?

### DISCUSSION THREAD

**Discussion by josephinoo** [PMEAL/OpenPNM#2786] [Q&A]
![image](https://github.com/PMEAL/porespy/assets/46098690/50abedd8-39ba-4ae0-ab87-7634bf2da6c4)

 Knowing that I have the inlet and outlet pores and knowing that the edge is the distance between the pores, there is a way to traverse them.

**Comment by jgostick** [comment]
This is probably an OpenPNM question.  Once you extract then network using PoreSpy, then all the 'network' calculations are done using the OpenPNM network. I am going to transfer this to that repo, then I'll give a bit of an answer.

**Reply by jgostick** [reply]
To follow-up my comment, once you have the network in OpenPNM you can use the adjacency matrix to do queries like the one you're asking about.  You can get the adjecency matrix using `am = pn.create_adjacency_matrix(...)`, where you can specify the weights for the connections, and the sparse format to return.  You can then pass `am` into any of the algorithms in `scipy.sparse.csgraph`, like the breadth-first search, or Djikstra's algorithm.  I'm not quite sure what you are hoping to learn, so I hope my suggestion was enough to get you started.
